scripts/signedbuildskadi.sh

```diff
@@ -1,274 +1,311 @@
+ GRAFANA_PASSWORD=$NGINX_PASSWORD
+}
+
+setup_credentials () {
+  # Set Credentials
+  SECRET_KEY=$(openssl rand -base64 32 |sha256sum | sed 's/ //g')
+  POSTGRES_USER="timesketch"
+  psql_pw=$(openssl rand -base64 32 |sha256sum | sed 's/ //g')
+  neo4juser='neo4j'
+
+  # Use dynamically created passwords with an option to use defaults
+  if [ $default_skadi_passwords = "false" ]
+    then
+      echo "Using random username and passwords for OS Account, TimeSketch, Nginx proxy / Grafana"
+      echo "Writing all credentials to /opt/skadi_credentials"
+      TIMESKETCH_USER="skadi_$(openssl rand -base64 3)"
+      TIMESKETCH_PASSWORD=$(openssl rand -base64 32 |sha256sum | sed 's/ //g')
+      NGINX_USER="skadi_$(openssl rand -base64 3)"
+      NGINX_PASSWORD=$(openssl rand -base64 32 |sha256sum | sed 's/ //g')
+      GRAFANA_USER=$NGINX_USER
+      GRAFANA_PASSWORD=$NGINX_PASSWORD
+      SKADI_PASS=$(openssl rand -base64 32 |sha256sum | sed 's/ //g')
+      SKADI_USER_HOME="/home/$SKADI_USER"
+      echo "  Proxy & Grafana Account:" > /opt/skadi_credentials
+      echo "     - Username: $NGINX_USER" >> /opt/skadi_credentials
+      echo "     - Password: $NGINX_PASSWORD" >> /opt/skadi_credentials
+      echo "" >> /opt/skadi_credentials
+      echo "  TimeSketch Account:" >> /opt/skadi_credentials
+      echo "     - Username: $TIMESKETCH_USER" >> /opt/skadi_credentials
+      echo "     - Password: $TIMESKETCH_PASSWORD" >> /opt/skadi_credentials
+  else
+      echo "Using Skadi default username and password of skadi:skadi for OS Account, TimeSketch, Nginx proxy, and Grafana"
+  fi
+}
+
+setup_docker () {
+  # Add Docker gpg key
+  curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+  sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+
+  # Install Docker
+  sudo apt-get update
+  sudo apt-get install docker-ce -y
+  sudo systemctl enable docker
+
+  # Install Docker-Compose
+  sudo curl -L "https://github.com/docker/compose/releases/download/1.23.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+  sudo chmod +x /usr/local/bin/docker-compose
+  sudo curl -L https://raw.githubusercontent.com/docker/compose/1.23.1/contrib/completion/bash/docker-compose -o /etc/bash_completion.d/docker-compose
+
+  # Add skadi to docker usergroup
+  sudo usermod -aG docker $SKADI_USER
+}
+
+cdqr_cylr_setup () {
+  # Clone skadi and it's submodules
+  sudo git clone --recurse-submodules --branch $install_branch https://github.com/orlikoski/Skadi.git /opt/Skadi
+  sudo chown -R $SKADI_USER:$SKADI_USER /opt/Skadi
+
+  # Copy cdqr script to /usr/local/bin
+  sudo cp /opt/Skadi/scripts/cdqr /usr/local/bin/cdqr
+  sudo chmod +x /usr/local/bin/cdqr
+
+  # Create CyLR directory
+  sudo mkdir /opt/CyLR/
+  sudo chmod 777 /opt/CyLR
+
+  # Installs and Configures CDQR and CyLR
+  sudo -E bash /opt/Skadi/scripts/update.sh
+}
+
+timesketch_configs () {

```

Comment:
this function should only execute when the default creds aren't used. We now load the default config file into the repo so no changes are needed if default creds are used.